src/RepositoryStore.php

```diff
@@ -0,0 +1,131 @@
+ * file that was distributed with this source code.
+ */
+
+namespace League\FactoryMuffin;
+
+use League\FactoryMuffin\Exceptions\DeleteMethodNotFoundException;
+use League\FactoryMuffin\Exceptions\DeletingFailedException;
+use League\FactoryMuffin\Exceptions\MethodNotFoundException;
+use League\FactoryMuffin\Exceptions\SaveMethodNotFoundException;
+
+/**
+ * This is the model store class.
+ *
+ * @author Michael Bodnarchuk <davert@codeception.com>
+ */
+class RepositoryStore extends ModelStore
+{
+    /**
+     * Methods to flush changes to storage.
+     *
+     * @var string
+     */
+    protected $flushMethod = 'flush';
+
+    /**
+     * Instance of a class that performs flushes (EntityManager for Doctrine).
+     *
+     * @var object
+     */
+    protected $storage;
+
+    public function __construct($storage)
+    {
+        $this->storage = $storage;
+        $this->saveMethod = 'persist';
+        $this->deleteMethod = 'remove';
+    }
+
+    /**
+     * Save our object to the db, and keep track of it.
+     *
+     * @param object $model The model instance.
+     *
+     * @throws \League\FactoryMuffin\Exceptions\SaveMethodNotFoundException
+     * @throws \League\FactoryMuffin\Exceptions\MethodNotFoundException
+     *
+     * @return bool
+     */
+    protected function save($model)
+    {
+        $method = $this->saveMethod;
+
+        if (!method_exists($this->storage, $method)) {
+            throw new SaveMethodNotFoundException(get_class($this->storage), $method);
+        }
+
+        $this->storage->$method($model);
+        $this->flush();
+
+        return true;
+    }
+
+    /**
+     * Flushes changes to storage.
+     *
+     * @throws \League\FactoryMuffin\Exceptions\MethodNotFoundException
+     *
+     * @return bool
+     */
+    protected function flush()
+    {
+        $method = $this->flushMethod;
+
+        if (!method_exists($this->storage, $method)) {
+            throw new MethodNotFoundException(get_class($this->storage), $method, "Can't save: flush method not found");
+        }
+
+        $this->storage->$method();
+
+        return true;
+    }
+
+    /**
+     * Delete our object from the db.
+     *
+     * @param object $model The model instance.
+     *
+     * @throws \League\FactoryMuffin\Exceptions\DeleteMethodNotFoundException
+     *
+     * @return bool
+     */
+    protected function delete($model)
+    {
+        $method = $this->deleteMethod;
+
+        if (!method_exists($this->storage, $method)) {
+            throw new DeleteMethodNotFoundException(get_class($this->storage), $method);
+        }
+
+        $this->storage->$method($model);
+
+        return true;
+    }
+
+    /**
+     * Delete all the saved models.
+     *
+     * @throws \League\FactoryMuffin\Exceptions\DeletingFailedException
+     *
+     * @return void
+     */
+    public function deleteSaved()
+    {
+        parent::deleteSaved();
+
+        try {
+            $this->storage->flush();
```

Comment:
@DavertMik I assume this was meant to be `$this->flush()`, otherwise we're ignoring the flushMethod property?